pkg/image/association.go

```diff
@@ -35,8 +35,18 @@ func (e *ErrInvalidComponent) Error() string {
 	return fmt.Sprintf("image %q has invalid component %q", e.image, e.tag)
 }
 
-// Associations is a set of image Associations.
-type Associations map[string]Association
+// AssociationSet is a mapped for Association
+// searching
+type AssociationSet map[string]Association
+
+// Associations is a set of image Associations
+// mapped to their images
+type Associations map[string]AssociationSet
+
+// NewAssociations instantiates a new multimap.
+func NewAssociations() Associations {
```

Comment:
I don't think this constructor is adding much over the bare `a := AssociationSet{}`, unless you intend on adding parameters at some point?

Reply:
That's a fair point. I don't expect that parameter will come into play at any point here.

Reply:
Resolving. Removed the NewAssociations function.